File: python class 12/Projects/stone paper scissors.py
import random

print("-----Stone Paper Scissors-----\n-----Press '69' any time to exit-----")
def with_comp():
    opt=['r','p','s']
    while True:
        comp_choice=random.choice(opt)
        user_choice=input("Rock(r) Paper(p) Scissors(s): ")
        if user_choice=='69':
            break

        elif comp_choice==user_choice:
            print('Draw!')
            continue
        
        elif comp_choice=='r' and user_choice=='p':
            print('You Win!')
            print(f'Computer\'s choice: {comp_choice}\n\n')            
            continue
        elif comp_choice=='r' and user_choice=='s':
            print('Computer Win!')
            print(f'Computer\'s choice: {comp_choice}\n\n')
            continue
        
        elif comp_choice=='p' and user_choice=='r':
            print('Computer Win!')
            print(f'Computer\'s choice: {comp_choice}\n\n')
            continue
        elif comp_choice=='p' and user_choice=='s':
            print('You Win!')
            print(f'Computer\'s choice: {comp_choice}\n\n')
            continue

        elif comp_choice=='s' and user_choice=='r':
            print('You Win!')
            print(f'Computer\'s choice: {comp_choice}\n\n')
            continue
        elif comp_choice=='s' and user_choice=='p':
            print('Computer Win!')
            print(f'Computer\'s choice: {comp_choice}\n\n')
            continue

def two_player():
    while True:
        user1_choice=input("Player 1:Rock(r) Paper(p) Scissors(s): ")
        user2_choice=input("Player 2:Rock(r) Paper(p) Scissors(s): ")

        if user1_choice=='69' or user2_choice=='69':
            break
        if user2_choice==user1_choice:
            print('Draw!')
            continue
        
        elif user2_choice=='r' and user1_choice=='p':
            print('Player 1 Win!\n\n')
            continue
        elif user2_choice=='r' and user1_choice=='s':
            print('Player 2 Win!\n\n')
            continue
        
        elif user2_choice=='p' and user1_choice=='r':
            print('Player 2 Win!\n\n')
            continue
        elif user2_choice=='p' and user1_choice=='s':
            print('Player 1 Win!\n\n')
            continue

        elif user2_choice=='s' and user1_choice=='r':
            print('Player 1 Win!\n\n')
            continue
        elif user2_choice=='s' and user1_choice=='p':
            print('Player 2 Win!\n\n')
            continue

choice=int(input("How do you wanna play?\nPress 1:With Computer.\nPress 2:Two Player Game."))
if choice==1:
    with_comp()
elif choice==2:
    two_player()

File: College C/stone_paper_scissors.c
/*Stone paper Scissors*/

#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <time.h>


int GenRandNum(int n){
    srand(time(NULL));
    return rand()%n;
}

int main()
{
    int compscore=0, userscore=0;
    char userchoice, comp_choice;

    printf("ROCK - PAPER - SCISSORS\n\n\tr-rock\n\tp-paper\n\ts-scissors\n\n");
    for (int i = 0; i < 3; i++)
    {
        printf("Rock/Paper/Scissors: ");
        scanf("%c",&userchoice);
        getchar();
        comp_choice=GenRandNum(3);//0--rock, 1--paper, 2--scissors
        if ((userchoice=='r' && comp_choice==0) || (userchoice=='p' && comp_choice==1)||(userchoice=='s' && comp_choice==2))
        {
            printf("It's a Draw!\n");
        }
        else if ((userchoice=='r' && comp_choice==1))
        {
            printf("Computer Wins!(Computer choice:paper)\n");
            compscore++;

        }
        else if ((userchoice=='r' && comp_choice==2))
        {
            printf("User wins(Computer choice:scissors)\n");
            userscore++;
        }
        else if ((userchoice=='p' && comp_choice==0))
        {
            printf("User Wins!(Computer choice:rock)\n");
            userscore++;
        }
        else if ((userchoice=='p' && comp_choice==2))
        {
            printf("Computer wins!(Computer choice:scissors)\n");
            compscore++;
        }
        else if ((userchoice=='s' && comp_choice==0))
        {
            printf("Computer wins!(Computer choice:rock)\n");
            compscore++;
        }
        else if ((userchoice=='s' && comp_choice==1))
        {
            printf("User wins!(Computer choice:paper)\n");
            userscore++;
        }
    }
    if (userscore==compscore)
    {
        printf("DRAW!!");
    }
    else if (userscore>compscore)
    {
        printf("\nUser wins the Game(Total points: %d)",userscore);
    }
    else{
        printf("\nComputer wins the Game(Total points: %d)",compscore);
    }
    


    return 0;
}